### ISSUE TITLE
set_dataset() cannot use relative dataset path

### ISSUE THREAD

**Issue by rosemarychen99** [pharmpy/pharmpy#3115]
### Bug report

Pharmpy version: 0.110.0

This code does not work `start.mod <- create_basic_pk_model(administration="oral") %>% 
  set_dataset("data.csv", datatype="nonmem")`

But this code works `start.mod <- create_basic_pk_model(administration="oral") %>% 
  set_dataset("/Users/xiach933/Work/Manuscript/AMD/example_code/data.csv", datatype="nonmem")`

**Comment by stellabelin** [comment]
Keeping this and marking #2961 as a duplicate since this has a clearer title.

**Comment by stellabelin** [comment]
Hi! I can't recreate this, what error do you get?

**Comment by rosemarychen99** [comment]
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  AssertionError

**Comment by rikardn** [comment]
I can reproduce the issue in Python:

```python
import os

os.chdir("mydir")
m1 = read_model("run1.mod")
m2  = set_dataset(model, "datafile", datatype="nonmem")
```

This gives:

```
Traceback (most recent call last):

  File "/tmp/ipykernel_6503/3833106283.py", line 1, in <module>
    m2 = set_dataset(model, "pheno.dta", datatype="nonmem")

  File "/home/rikard/devel/pharmpy/src/pharmpy/modeling/data.py", line 2261, in set_dataset
    di = update_datainfo(di, df).replace(path=path)

  File "/home/rikard/devel/pharmpy/src/pharmpy/model/datainfo.py", line 603, in replace
    return DataInfo.create(columns=columns, path=path, separator=separator)

  File "/home/rikard/devel/pharmpy/src/pharmpy/model/datainfo.py", line 587, in create
    assert not force_absolute_path or path is None or path.is_absolute()

AssertionError
```

**Comment by rikardn** [comment]
Fixed with b0df15b6083d73d82e29107ff9929c23b7039293. Thanks for reporting!
